src/code/code_80140E80.c

```diff
@@ -1,21 +1,359 @@
+/**
+ * @file code_80140E80.c
+ * @brief Copies images between color images (generally framebuffers), possibly with filling, scaling, and colour
+ * interpolation.
+ *
+ * Used for several transition effects, and in z_play to shrink the screen at the end of the First and Second Days.
+ *
+ *
+ *
+ * @note to use the functions in this file, with the exception of func_80141778(), it is necessary to load the S2DEX2
+ * microcode first, and then re-load the 3D microcode afterwards for the rest of the drawing in the frame; see
+ * func_80141778() for how to do this.
+ */
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_80140E80/func_80140E80.s")
+// z64.h once named properly
+typedef enum {
+    /* 0 */ BG_CYC_1CYC, // Enable rescaling
+    /* 1 */ BG_CYC_COPY  // Direct copy, no scaling
+} BgMode;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_80140E80/func_80140EA0.s")
+typedef enum {
+    /* 0 */ FB_MODE_DO_NOTHING, // Do nothing but waste time loading microcode
+    /* 1 */ FB_MODE_GENERAL,    // Interpolation, filling and scaling
+    /* 2 */ FB_MODE_INTERPOLATE
+} FbMode;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_80140E80/func_80140EAC.s")
+// ucode.h
+#define SP_UCODE_DATA_SIZE 0x800
```

Comment:
Could we put together `ucode.h`? I know this same define is needed for `Interface_Draw`